src/patroni.py

```diff
@@ -48,6 +51,14 @@ def __init__(
         self._planned_units = planned_units
         self._superuser_password = superuser_password
         self._replication_password = replication_password
+        self._tls_enabled = tls_enabled
+        # Variable mapping to requests library verify parameter.
+        self._verify = f"{self._storage_path}/{TLS_CA_FILE}" if tls_enabled else True
```

Comment:
If I understand correctly, `requests.get(verify=True)` using the local public root certificate. Where is that in this case?

Reply:
Shouldn't it be `verify=False if not tls_enabled`?

Reply:
I added a better comment on https://github.com/canonical/postgresql-k8s-operator/pull/38/commits/9ff5c810f464af161bc0345696f48226ee381ce5. The `verify` flag is set to the CA bundle file when TLS is enabled to correctly verify the API server certificates. When TLS is disabled, the charm set the True value (which Is the [default](https://github.com/psf/requests/blob/2a6f290bc09324406708a4d404a88a45d848ddf9/requests/api.py#L39) from the library), which seems to not validate anything when TLS is disabled (and we are requesting something from an http url).